cmd/query/app/server.go

```diff
@@ -131,7 +157,33 @@ func createGRPCServer(querySvc *querysvc.QueryService, metricsQuerySvc querysvc.
 	healthServer.SetServingStatus("jaeger.api_v3.QueryService", grpc_health_v1.HealthCheckResponse_SERVING)
 
 	grpc_health_v1.RegisterHealthServer(server, healthServer)
-	return server, nil
+}
+
+func createGRPCServerOTEL(
+	ctx context.Context,
+	options *QueryOptions,
+	tm *tenancy.Manager,
+	telset telemetery.Setting,
+) (*grpc.Server, error) {
+	var grpcOpts []configgrpc.ToServerOption
```

Comment:
let's consolidate this part across two implementations. We are also missing bearer token interceptors being introduced in #6063

Reply:
@yurishkuro How do you want to go about this? the two implementations have different options. For example - the legacy server loads the TLS configuration but the new implementation handles this inside OTEL

Reply:
I was only referring to `grpcOpts` which only contain tenancy interceptors.

Reply:
@yurishkuro in the legacy implementation - the `grpcOpts` also contains TLS credentials
```go
	var grpcOpts []grpc.ServerOption

	if options.GRPC.TLSSetting != nil {
		tlsCfg, err := options.GRPC.TLSSetting.LoadTLSConfig(ctx)
		if err != nil {
			return nil, err
		}

		creds := credentials.NewTLS(tlsCfg)

		grpcOpts = append(grpcOpts, grpc.Creds(creds))
	}
```

Reply:
I was only referring to tenancy interceptors and later bearer token interceptors from #6063

Reply:
actually, never mind, the types of options are different.